src/wallet/rpcdump.cpp

```diff
@@ -433,7 +438,9 @@ UniValue importpubkey(const JSONRPCRequest& request)
 
     LOCK2(cs_main, pwallet->cs_wallet);
 
-    ImportAddress(pwallet, pubKey.GetID(), strLabel);
+    for (const auto& dest : GetAllDestinationsForKey(pubKey)) {
+        ImportAddress(pwallet, dest, strLabel);
```

Comment:
Is it safe that we do not do a AddCScript based on a importpubkey? Seems to me that you'd never match the HaveCScript condition if you downgrade so maybe something might break? The final script will still be watched, but it just seems weird to not have the symmetry with importprivkey here.

Reply:
I believe you're right. Fixed.